media/js/base/banners/mozilla-banner.es6.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+import { getConsentCookie } from '../consent/utils.es6';
+
+let _pageBanner;
+const MozBanner = {};
+
+MozBanner.consentsToCookies = function () {
+    const cookie = getConsentCookie();
+
+    if (cookie) {
+        return cookie.preference;
+    } else {
+        return true;
+    }
+};
+
+MozBanner.setCookie = function (id) {
+    /**
+     * When closing a banner we set a "preference cookie" to remember
+     * that a user no longer wishes to see it on repeat page visits.
+     * Legal are OK to set this cookie without explicit consent in
+     * EU/EAA countries because:
+     *
+     * 1) The cookie is not used for tracking purposes in any way.
+     * 2) The cookie is set only after an explicit action by the user
+     *    that signals clear intent.
+     *
+     * We still honor not setting this cookie if preference cookies
+     * have been explicitly rejected by the user.
+     */
+    if (!MozBanner.consentsToCookies()) {
+        return;
+    }
```

Comment:
Note for reviewer: most of this file is just existing code that has been reformatted to modern JS. The only real new code added is here.